### ISSUE TITLE
[bug] `immutability`: false positive on writes through a `RefObject` prop

### ISSUE THREAD

**Issue by silverwind** [Rel1cx/eslint-react#1751]
*This issue was written by Claude, an AI assistant.*

## Rule

`react/immutability`

## Description

The rule flags writes through a `RefObject<T>` received as a prop, even though `RefObject` is intentionally mutable by design (its `.current` is meant to be reassigned, and the referenced DOM element is meant to be mutated). Identical writes through a `useRef`-created ref in the same component are not flagged.

## Repro

```tsx
import type {RefObject} from "react";

type Props = {inputRef: RefObject<HTMLInputElement | null>};

export function ChildA({inputRef}: Props) {
  const callbackRef = (node: HTMLInputElement | null) => {
    inputRef.current = node; // false-positive
  };
  return <input ref={callbackRef} />;
}

export function ChildB({inputRef}: Props) {
  function reset() {
    if (inputRef.current) inputRef.current.value = ""; // false-positive
  }
  return <input ref={inputRef} onBlur={reset} />;
}
```

Versions: `eslint-plugin-react-x@5.7.2`, `eslint@10.3.0`, `@typescript-eslint/parser@8.59.2`, `typescript@6.0.3`, `react@19.2.5`.

## Expected

No error in `ChildA`/`ChildB`. `RefObject<T>` is mutable by API contract — assigning to `.current` is the standard ref-forwarding pattern, and mutating DOM properties through `.current` is normal imperative DOM access (same case as #1626 but via a ref rather than `e.currentTarget`).

## Actual

```
Do not mutate 'inputRef' directly. Props and state are immutable — create a new object instead          react/immutability
Do not mutate 'inputRef.current' directly. Props and state are immutable — create a new object instead  react/immutability
```
